subsys/math/public/cmsis_compiler.h

```diff
@@ -0,0 +1,65 @@
+/* Copyright (c) 2022 Google LLC
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+/**
+ * @file cmsis_compiler.h
+ * @brief Basic shim layer to get cmsis-dsp working on non arm platforms
+ *
+ * This file is used when CMSIS-Core is not available.
+ */
+
+#ifndef SUBSYS_MATH_PUBLIC_CMSIS_COMPILER_H_
+#define SUBSYS_MATH_PUBLIC_CMSIS_COMPILER_H_
+
+#include <zephyr/sys/util.h>
+
+/* Map the CMSIS __ALIGNED to the Zephyr __aligned macro */
+#define __ALIGNED(x) __aligned(x)
+
+/* Make sure that ALWAYS_INLINE is defined (at least to just be inline) */
+#ifndef ALWAYS_INLINE
+#define ALWAYS_INLINE inline
+#endif
+
+/* Define the CMSIS __STATIC_FORCEINLINE to use Zephyr's macros */
+#define __STATIC_FORCEINLINE static ALWAYS_INLINE
+
+/* Define the CMSIS __STATIC_INLINE to a plain 'static inline' */
+#define __STATIC_INLINE static inline
+
+/**
+ * @brief Signed Saturate
+ *
+ * Saturates a signed value.
+ *
+ * @param ARG1 Value to be saturated
+ * @param ARG2 Bit position to saturate to (1..32)
+ * @return Saturated value
+ */
+#define __SSAT(ARG1, ARG2)                                                                         \
```

Comment:
This needs to be defined only for aarch64.

Reply:
Currently, with the `HAS_CMSIS_CORE` as is, AARCH32 will select CoreA or CoreR from the CMSIS module. Can you explain in what case do we not need this?

Reply:
> Currently, with the `HAS_CMSIS_CORE` as is, AARCH32 will select CoreA or CoreR from the CMSIS module. Can you explain in what case do we not need this?

non-ARM architectures?

Also for AArch64, [the CMSIS-DSP doc says you should define `__GNUC_PYTHON__`](https://github.com/ARM-software/CMSIS-DSP#how-to-build-for-aarch64), in which case this header file probably will not be necessary at all.

The same should apply to other non-ARM architectures without CMSIS-Core.

Reply:
FYI, https://github.com/zephyrproject-rtos/cmsis/blob/093de61c2a7d12dc9253daf8692f61f793a9254a/CMSIS/DSP/CMakeLists.txt#L12

Reply:
Thanks, that works. I'm getting one issue with `qemu_nios2` giving me `SKIPPED (FLASH overflow)`. Can you point me at an example of extending the flash space on the qemu?